utils/terrascript_client.py

```diff
@@ -1485,6 +1492,163 @@ def populate_tf_resource_cloudwatch(self, resource, namespace_info):
             values['provider'] = 'aws.' + region
         log_group_tf_resource = aws_cloudwatch_log_group(identifier, **values)
         tf_resources.append(log_group_tf_resource)
+
+        es_identifier = common_values.get('es_identifier', None)
+        if es_identifier is not None:
+
+            assume_role_policy = {
+                "Version": "2012-10-17",
+                "Statement": [
+                    {
+                        "Action": "sts:AssumeRole",
+                        "Principal": {
+                            "Service": "lambda.amazonaws.com"
+                        },
+                        "Effect": "Allow"
+                    }
+                ]
+            }
+
+            role_identifier = f"{identifier}-lambda-execution-role"
+            role_values = {
+                'name': role_identifier,
+                'assume_role_policy':
+                    json.dumps(assume_role_policy, sort_keys=True)
+            }
+
+            role_tf_resource = aws_iam_role(role_identifier, **role_values)
+            tf_resources.append(role_tf_resource)
+
+            policy_identifier = f"{identifier}-lambda-execution-policy"
+            policy = {
+                "Version": "2012-10-17",
+                "Statement": [
+                    {
+                        "Effect": "Allow",
+                        "Action": [
+                            "logs:CreateLogGroup",
+                            "logs:CreateLogStream",
+                            "logs:PutLogEvents"
+                        ],
+                        "Resource": "*"
+                    },
+                    {
+                        "Effect": "Allow",
+                        "Action": "es:*",
+                        "Resource": "*"
+                    },
+                    {
+                        "Effect": "Allow",
+                        "Action": [
+                            "ec2:DescribeNetworkInterfaces",
+                            "ec2:CreateNetworkInterface",
+                            "ec2:DeleteNetworkInterface",
+                            "ec2:DescribeInstances",
+                            "ec2:AttachNetworkInterface"
+                        ],
+                        "Resource": "*"
```

Comment:
Can we list specific resources instead of `*`?

Reply:
Find an existing role policy provided by AWS. https://console.aws.amazon.com/iam/home?region=us-east-1#/policies/arn:aws:iam::aws:policy/service-role/AWSLambdaVPCAccessExecutionRole$jsonEditor. and they use "*"

Reply:
Lambda need this policy to access resources in VPC according to https://docs.aws.amazon.com/lambda/latest/dg/configuration-vpc.html